zebra-chain/src/transaction.rs

```diff
@@ -152,4 +154,14 @@ impl Transaction {
                 Some(transparent::Input::Coinbase { .. })
             )
     }
+
+    /// Calculate the sighash for the current transaction
+    pub fn sighash(
```

Comment:
Since this is the public interface to sighash, let's document each argument.

In particular, it's unclear what the u32 in the input argument is.
Do we need a struct here, rather than a tuple?
Then each argument would have a name.

Also, this function panics if the network upgrade is Genesis or BeforeOverwinter, or if the transaction is V1 or V2.